pkg/k8sutil/status.go

```diff
@@ -43,6 +44,58 @@ func IsMarkedForDeletion(m metav1.ObjectMeta) bool {
 	return m.GetDeletionTimestamp() != nil
 }
 
+// UpdateBrokerConfigurationBackup updates the broker status with a backup from kafka broker configurations
+func UpdateBrokerConfigurationBackup(c client.Client, cluster *banzaicloudv1beta1.KafkaCluster) error {
+	needsUpdate, err := generateBrokerConfigurationBackups(cluster)
+	if err != nil {
+		return err
+	}
+	if !needsUpdate {
+		return nil
+	}
+	ctx := context.Background()
+	if err := c.Status().Update(ctx, cluster); err != nil {
+		if !apierrors.IsConflict(err) {
+			return errors.WrapIff(err, "could not update Kafka broker(s) configuration backup state")
+		}
+		if err := c.Get(ctx, types.NamespacedName{Namespace: cluster.Namespace, Name: cluster.Name}, cluster); err != nil {
+			return errors.WrapIf(err, "could not get config for updating status")
+		}
+		needsUpdate, err := generateBrokerConfigurationBackups(cluster)
+		if err != nil {
+			return err
+		}
+		if !needsUpdate {
+			return nil
+		}
+		if err = c.Status().Update(ctx, cluster); err != nil {
+			return errors.WrapIff(err, "could not update Kafka broker(s) configuration backup state")
+		}
+	}
+	return nil
+}
+
+func generateBrokerConfigurationBackups(cluster *banzaicloudv1beta1.KafkaCluster) (bool, error) {
+	needsUpdate := false
+	if cluster.Status.BrokersState == nil {
+		cluster.Status.BrokersState = make(map[string]banzaicloudv1beta1.BrokerState)
+	}
+
+	for _, broker := range cluster.Spec.Brokers {
+		brokerState := cluster.Status.BrokersState[fmt.Sprint(broker.Id)]
+		configurationBackup, err := util.GzipAndBase64BrokerConfiguration(&broker)
```

Comment:
This would increase the size of KafkaCluster custom resource. Do you see a risk of hitting the 1MB limit overall for large kafka clusters (say hundreds of brokers)?

Reply:
It takes ~500bytes/broker 500x100 = 50Kbytes.
Configuration can be stored also in configmap but I think it is not so safe.

Reply:
When different configurations are used with lot of configuration entry for many brokers there is a chance to reach the 1MB. I think it is a very edge case.

Reply:
sounds good then

Reply:
I realised that you store only the broker node and not the whole broker expanded config.

In our case - we drive the broekr config through brokerConfigGroup(s) and thus the broker entry is quite terse, hence the size here very small:
```
  brokers:
  - brokerConfigGroup: az1
    id: 1101
    readOnlyConfig: |
      broker.rack=eastus2-1
```